collections/README.md

```diff
@@ -867,15 +863,23 @@ Where the first type parameter is `uint64`, which is the field type of our index
 The second type parameter is the primary key `sdk.AccAddress`
 And the third type parameter is the actual object we're storing `authtypes.BaseAccount`.
 
-Then we implement a function called `IndexesList` on our `AccountIndexes` struct, this will be used
-by the `IndexedMap` to keep the underlying map in sync with the indexes, in our case `Number`.
-This function just needs to return the slice of indexes contained in the struct.
-
 Then we create a `NewAccountIndexes` function that instantiates and returns the `AccountsIndexes` struct.
 
 The function takes a `SchemaBuilder`. Then we instantiate our `indexes.Unique`, let's analyse the arguments we pass to
 `indexes.NewUnique`.
 
+#### NOTE: indexes list
+
+The `AccountsIndexes` struct contains the indexes, the `NewIndexedMap` function will infer the indexes form that struct
+using reflection, this happens only at init and is not computationally expensive. In case you want to explicitly declare
+indexes: implement the `Indexes` interface in the `AccountsIndexes` struct:
```

Comment:
What are the cases in which a user would like to define these? Or are we leaving it just for compatibility?

Reply:
Compatibility and also because reflection inferring is really strict as it does not allow private fields, etc. 

Wanted to leave some degree of flexibility.